src/coreclr/jit/morph.cpp

```diff
@@ -16004,6 +16004,14 @@ void Compiler::fgMorphBlocks()
 
 #endif
 
+    if (!compEnregLocals())
+    {
+        // Morph is checking if lvDoNotEnregister is already set for some optimizations.
+        // If we are running in min opts and know that we won't enregister any locals
+        // it is better to set this flag before we start reading it.
```

Comment:
I was hoping to see MinOpts check here...why is it not there?

Reply:
The comment was poorly written, what I was trying to explain here:
1. the main flag is `(opts.compFlags & CLFLG_REGVAR)`, if it is not set we can't enregister variables;
2. right now it is set if we use `CLFLG_MAXOPT` and not set when using `CLFLG_MINOPT`, an example where we  set it:
https://github.com/dotnet/runtime/blob/7f88911fe5f89836d1de7efd23f8541514a5ed05/src/coreclr/jit/compiler.cpp#L2615-L2620

3. in theory nothing prevents us to disable CLFLG_REGVAR when MinOpts is not set, or enable it when it is set, so I do not want to check `isMinOpts` instead of `!compEnregLocals()` here.

I have updated the comments to reflect that, please take a look.